### PR TITLE
Discovery of GATTs

### PR DESCRIPTION
**Pull request by mikolak**
Added services, characteristics and descriptors to example simulated peripherals

Polidea/react-native-blemulator#18

### CODE REVIEW

File: src/Saga.js [hunk 1]
```diff
@@ -240,7 +280,7 @@ function* handleConnection(manager: BleManager): Generator<*, *, *> {
       yield put(updateConnectionState(ConnectionState.CONNECTING));
       yield call([device, device.connect]);
       yield put(updateConnectionState(ConnectionState.DISCOVERING));
-    //   yield call([device, device.discoverAllServicesAndCharacteristics]); // TODO uncomment
+      yield call([device, device.discoverAllServicesAndCharacteristics]); // TODO uncomment
```
Comment (mikolak) [inline src/Saga.js]:
```suggestion
      yield call([device, device.discoverAllServicesAndCharacteristics]);
```

File: src/Tests.js [hunk 2]
```diff
@@ -52,14 +52,14 @@ function* readAllCharacteristics(device: Device): Generator<*, boolean, *> {
 
         for (const descriptor of descriptors) {
           yield put(log('* Found descriptor: ' + descriptor.uuid));
-          const d: Descriptor = yield call([descriptor, descriptor.read]);
-          yield put(log('Descriptor value: ' + (d.value || 'null')));
-          if (d.uuid === '00002902-0000-1000-8000-00805f9b34fb') {
-            yield put(log('Skipping CCC'));
-            continue;
-          }
+        //   const d: Descriptor = yield call([descriptor, descriptor.read]);
```
Comment (pawelByszewski) [inline src/Tests.js:55]: why is it commented out instead of removed?

File: src/Tests.js [hunk 3]
```diff
@@ -52,14 +52,14 @@ function* readAllCharacteristics(device: Device): Generator<*, boolean, *> {
...
+        //   yield put(log('Descriptor value: ' + (d.value || 'null')));
+        //   if (d.uuid === '00002902-0000-1000-8000-00805f9b34fb') {
+        //     yield put(log('Skipping CCC'));
+        //     continue;
+        //   }
           try {
-            yield call([descriptor, descriptor.write], 'AAA=');
+            // yield call([descriptor, descriptor.write], 'AAA=');
```
Comment (pawelByszewski) [inline src/Tests.js:62]: why is it commented out instead of removed?

File: src/Tests.js [hunk 4]
```diff
@@ -73,13 +73,13 @@ function* readAllCharacteristics(device: Device): Generator<*, boolean, *> {
         yield put(log('Found characteristic: ' + characteristic.uuid));
         if (characteristic.isReadable) {
           yield put(log('Reading value...'));
-          var c = yield call([characteristic, characteristic.read]);
-          yield put(log('Got base64 value: ' + c.value));
+        //   var c = yield call([characteristic, characteristic.read]);
```
Comment (pawelByszewski) [inline src/Tests.js:76]: why is it commented out instead of removed?

File: src/Tests.js [hunk 5]
```diff
@@ -73,13 +73,13 @@ function* readAllCharacteristics(device: Device): Generator<*, boolean, *> {
...
+        //   yield put(log('Got base64 value: ' + c.value));
           if (characteristic.isWritableWithResponse) {
-            yield call(
-              [characteristic, characteristic.writeWithResponse],
-              c.value,
-            );
+            // yield call(
```
Comment (pawelByszewski) [inline src/Tests.js:79]: why is it commented out instead of removed?

File: src/Tests.js (hunk 2 above)
Comment (mikolak) [reply src/Tests.js:55]: Read/write operation on Descriptors and Characteristics are not yet implemented, so the test failed. They'll be gradually reintroduced as we roll out next features.

File: src/Tests.js (hunk 3 above)
Comment (mikolak) [reply src/Tests.js:62]: Read/write operation on Descriptors and Characteristics are not yet implemented, so the test failed. They'll be gradually reintroduced as we roll out next features.

File: src/Tests.js (hunk 4 above)
Comment (mikolak) [reply src/Tests.js:76]: Read/write operation on Descriptors and Characteristics are not yet implemented, so the test failed. They'll be gradually reintroduced as we roll out next features.

File: src/Tests.js (hunk 5 above)
Comment (mikolak) [reply src/Tests.js:79]: Read/write operation on Descriptors and Characteristics are not yet implemented, so the test failed. They'll be gradually reintroduced as we roll out next features.

**Review by pawelByszewski — APPROVED** [review]

**Pull request merged**